app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/NotificationFilterActivity.java

```diff
@@ -0,0 +1,214 @@
+
+                // TODO: check for modifications, only save if something changed
+                String words = mEditTextWords.getText().toString();
+
+                if (StringUtils.isBlank(words) && mSpinnerFilterMode.getSelectedItemPosition() != NOTIFICATION_FILTER_MODE_NONE) {
+                    Toast.makeText(NotificationFilterActivity.this, R.string.toast_notification_filter_words_empty_hint, Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                try {
+                    db = GBApplication.acquireDB();
+                    NotificationFilterDao notificationFilterDao = db.getDaoSession().getNotificationFilterDao();
+                    NotificationFilterEntryDao notificationFilterEntryDao = db.getDaoSession().getNotificationFilterEntryDao();
+
+                    debugOutput(notificationFilterDao);
+
+                    mNotificationFilter.setNotificationFilterMode(mSpinnerFilterMode.getSelectedItemPosition());
+                    mNotificationFilter.setNotificationFilterSubMode(mSpinnerFilterSubMode.getSelectedItemPosition());
+
+                    notificationFilterEntryDao.deleteByKeyInTx(mFilterEntryIds);
+
+                    Long filterId = notificationFilterDao.insertOrReplace(mNotificationFilter);
+
+                    // only save words if filter mode != none
+                    if (mNotificationFilter.getNotificationFilterMode() != NOTIFICATION_FILTER_MODE_NONE) {
+                        String[] wordsSplitted = words.split("\n");
+                        for (String temp : wordsSplitted) {
+
+                            if (StringUtils.isBlank(temp)) {
+                                continue;
+                            }
+
+                            temp = temp.trim();
+                            NotificationFilterEntry notificationFilterEntry = new NotificationFilterEntry();
+                            notificationFilterEntry.setNotificationFilterContent(temp);
+                            notificationFilterEntry.setNotificationFilterId(filterId);
+                            notificationFilterEntryDao.insert(notificationFilterEntry);
+                        }
+                    }
+
+                    Toast.makeText(NotificationFilterActivity.this, R.string.toast_notification_filter_saved_successfully, Toast.LENGTH_SHORT).show();
+                    NotificationFilterActivity.this.finish();
+
+                } catch (GBException e) {
+                    LOG.error("Could not acquire DB.", e);
+                    Toast.makeText(NotificationFilterActivity.this, "Database Error: " + e.getMessage(), Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+    }
+
+    private void debugOutput(NotificationFilterDao notificationFilterDao) {
+        if (BuildConfig.DEBUG) {
+
+            List<NotificationFilter> filters = notificationFilterDao.loadAll();
+
+            LOG.info(TAG, "Saved filters");
```

Comment:
Why do you use the Android Log here instead of the Logger instance? The latter logs both to the Android Log as well as a logfile, if enabled in the preferences.